src/commands/mod.rs

```diff
@@ -580,14 +582,32 @@ impl DaemonControl {
         }
     }
 
-    pub fn list_spend(&self) -> ListSpendResult {
+    pub fn list_spend(
+        &self,
+        txids: Option<Vec<bitcoin::Txid>>,
+    ) -> Result<ListSpendResult, CommandError> {
+        if let Some(ids) = &txids {
+            if ids.is_empty() {
+                return Err(CommandError::EmptyFilterList);
+            }
+        }
+
         let mut db_conn = self.db.connection();
-        let spend_txs = db_conn
-            .list_spend()
+        let spend_psbts = db_conn.list_spend();
```

Comment:
I'm not sure if this has been discussed, but would it be worth also adding a filter on the DB query in `list_spend`? Perhaps at least if the number of txids is not too large so that the query string is not too long.

Reply:
you mean in order to apply the filtering at the DB level?

Reply:
That's right. We did a similar thing for listing coins by status and/or outpoint.

Reply:
not strong idea, but i guess it should be done at only one place to avoid duplicate 'filtering feature', i can move the filtering to DB level if required